src/sparseml/pytorch/optim/quantization/helpers.py

```diff
@@ -37,15 +39,40 @@
 ]
 
 
+_QUANTIZABLE_MODULE_TYPES = (
+    {
+        # Conv based layers
+        torch.nn.Conv1d,
+        torch.nn.Conv2d,
+        torch.nn.Conv3d,
+        nni.ConvBn1d,
+        nni.ConvBn2d,
+        nni.ConvBn3d,
+        nni.ConvReLU1d,
+        nni.ConvReLU2d,
+        nni.ConvReLU3d,
+        nni.ConvBnReLU1d,
+        nni.ConvBnReLU2d,
+        nni.ConvBnReLU3d,
+        # Linear Layers
+        torch.nn.Linear,
+        nni.LinearReLU,
+    }
+    if nni
+    else None
+)
+
+
 def add_quant_dequant(module):
     """
     Wraps all Conv and Linear submodule with a qconfig with a QuantWrapper
     :param module: the module to modify
     """
-    module_type = str(type(module)).split(".")[-1].lower()
-    is_quantizable_module = "conv" in module_type or "linear" in module_type
-
-    if is_quantizable_module and hasattr(module, "qconfig") and module.qconfig:
+    if (
+        type(module) in _QUANTIZABLE_MODULE_TYPES
```

Comment:
When does `nni` fail to import? And in that case `_QUANTIZABLE_MODULE_TYPES` would be `None`, and this check `type(module) in None` will fail to compile.

Reply:
the `nni` submodule isn't available on lower versions of PyTorch. our `optim.quantization` submodule isn't meant to be used with these lower versions, anyone using this function externally would already be using a compatible torch version, and internally, there are already version checks before this function is called.

Reply:
NIT: add some comment as to why nni won't fail to import